src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -274,25 +274,29 @@ public ShopifyVariant getVariant(final String variantId) {
 		return shopifyVariantRootResponse.getVariant();
 	}
 
-	public ShopifyProducts getProducts() {
+	public ShopifyProducts getProducts(final int pageSize) {
```

Comment:
I was just trying not to make this a major version change hence why I overloaded the query methods to add page size.

I could add another `getProducts` overload that takes in page and pageSize which does loop through all products but I will need to keep `getProducts()` if this is to remain a minor version change.

We will need to indicate pageSize either way as a parameter since we are trying to accommodate Shopify Internal Errors by dynamically adjusting page size. I wouldn't want to hotfix this again because Shopify Support is now suggesting we set a limit of 25 for the products query instead of 50.

Reply:
@ryankazokas do you think if I add another overload to `getProducts` that takes in `page` and `pageSize` foregoing the internal loop that would suffice here?

Reply:
Yeah, sorry. i didn't mean to remove 

> getProducts()

I just meant to add another param to the new method or make it so that it makes it different than the getProducts() method.  All i'm saying is that if the client wants to control page size, they probably want the granular control of paginating. Can't we just make the default 25 or a low number and add the new page to the new method so we'd have something like this:

`
public ShopifyProducts getProducts() {
   //does pagination for you with default page size using below method.
}

public ShopifyProducts getProducts(final int page, final int pageSize) {
    // client controls pagination
}
`

This doesn't introduce breaking changes, and provides two methods that provide two different outputs

Reply:
@ryankazokas awh ok ya I didn't get that from your initial comment. Sure I can accommodate that. I'll update the PR soon.